src/main/kotlin/org/hyacinthbots/lilybot/extensions/events/AutoThreading.kt

```diff
@@ -0,0 +1,529 @@
+				content = "Auto-threading has been **enabled** in this channel."
+			}
+
+		return interaction.textInputs["message"]!!.value!!
+	}
+
+	/**
+	 * A single function for both Proxied and Non-Proxied message to be turned into threads.
+	 *
+	 * @param event The event for the message creation
+	 * @param message The original message that wasn't proxied
+	 * @param proxiedMessage The proxied message, if the message was proxied
+	 * @since 4.4.0
+	 * @author NoComment1105
+	 */
+	private suspend fun <T : PKMessageCreateEvent> onMessageSend(
+		event: T,
+		message: Message?,
+		proxiedMessage: PKMessage? = null
+	) {
+		val memberFromPk = if (proxiedMessage != null) event.getGuild().getMember(proxiedMessage.sender) else null
+
+		val channel: TextChannel = if (proxiedMessage == null) {
+			message?.channel?.asChannelOf() ?: return
+		} else {
+			// Check the real message member too, despite the pk message not being null, we may still be able to use the original
+			message?.channel?.asChannelOf() ?: event.getGuild().getChannelOf(proxiedMessage.channel) ?: return
+		}
+
+		val authorId: Snowflake = if (proxiedMessage == null) {
+			message?.author?.id ?: return
+		} else {
+			message?.author?.id ?: proxiedMessage.sender
+		}
+
+		val options = AutoThreadingCollection().getSingleAutoThread(channel.id) ?: return
+
+		var threadName: String? = event.message.content.trim().split("\n").firstOrNull()?.take(75)
+
+		if (!options.contentAwareNaming || threadName == null) {
+			threadName = "Thread for ${
+				message?.author?.asUser()?.username ?: proxiedMessage?.member?.name
+			}".take(75)
+		}
+
+		if (options.preventDuplicates) {
+			var previousUserThread: TextChannelThread? = null
+			val ownerThreads = ThreadsCollection().getOwnerThreads(authorId)
+
+			val threadData = ownerThreads.find { it.parentChannelId == channel.id }
+			if (threadData != null) {
+				previousUserThread = event.getGuild().getChannelOfOrNull(threadData.threadId)
+			}
+
+			if (previousUserThread != null) {
+				val response = event.message.respond {
+					content = "Please use your existing thread in this channel ${previousUserThread.mention}"
+				}
+				event.message.delete("User already has a thread")
+				response.delete(10000L, false)
+				return
+			}
+		}
+
+		val thread = channel.startPublicThreadWithMessage(
+			message?.id ?: proxiedMessage!!.channel,
+			threadName,
+			channel.data.defaultAutoArchiveDuration.value ?: ArchiveDuration.Day
+		)
+
+		ThreadsCollection().setThreadOwner(event.getGuild().id, thread.id, event.member!!.id, channel.id)
+
+		handleThreadCreation(
+			options,
+			thread,
+			message?.author ?: memberFromPk!!.asUser()
+		)
+	}
+
+	private suspend inline fun handleThreadCreation(
+		inputOptions: AutoThreadingData,
+		inputThread: TextChannelThread,
+		inputUser: User
+	) {
+		val threadMessage = inputThread.createMessage(
+			if (inputOptions.mention) {
+				inputUser.mention
+			} else {
+				"Placeholder message"
+			}
```

Comment:
```suggestion
			if (inputOptions.mention) inputUser.mention else "Placeholder message"
```
Just a thought, it doesn't particularly need to take up 5 lines imo